app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,208 @@
+from core.models import Recipe, Ingredient
+from django.test import TestCase
+from django.urls import reverse
+from recipe.serializers import RecipeSerializer
+from rest_framework import status
+from rest_framework.test import APIClient
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def recipe_url(recipe_id):
+    """Return URL for a recipe"""
+
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def sample_recipe(params={}):
+    """Create and return a sample recipe"""
+
+    defaults = {
+        'name': 'Spaghetti Carbonara',
+        'description': 'A lovely meal',
+    }
+    defaults.update(params)
+
+    return Recipe.objects.create(**defaults)
+
+
+def sample_ingredient(recipe, name='Spaghetti'):
+    """Create and return a sample ingredient"""
+
+    return Ingredient.objects.create(recipe=recipe, name=name)
+
+
+class RecipeApiTests(TestCase):
+    """Test Recipe API endpoints"""
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_get_all_recipes(self):
+        """GET /recipes/ - Test returning a list of recipes"""
+
+        sample_recipe()
+        recipe = sample_recipe({
+            'name': 'Bangers and Mash',
+            'description': 'A hearty meal'
+        })
+
+        recipe.ingredients.add(
+            sample_ingredient(recipe=recipe, name='Sausage'))
+
+        all_recipes = Recipe.objects.all().order_by('id')
+        serializer = RecipeSerializer(all_recipes, many=True)
+
+        response = self.client.get(RECIPES_URL)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+        self.assertEqual(response.data, serializer.data)
```

Comment:
Could you compare with the actually data? Since the serializer is part of the API the comparison doesn't make too much sense. Or better said, you're just testing the view, not the serializer. Fine, if this is what you wanted.